A couple improvements

Lennix opened the pull request:
Created RpcClient.Startup which implements client-like behavior

Moved start of heartbeatDispatcher to StartDispatcher to start it after Startup is complete (and EventHandlers subscribed)

Use _session.GlobalSettingHash for DownloadSettingMessage requests

Lennix committed:
Use _session.GlobalSettingHash for DownloadSettingMessage if available
Send empty payload if not available (client behavior)

Lennix committed:
Possible request for GetIncensePokemon
(Added to every request when Incense is active)

Lennix committed:
Send Request for Startup
- GetPlayer
- DownloadRemoteConfig
- AssetDigest
- LevelupRewards
(Client behavior)

Lennix committed:
Added GetPokemonSortedByDistance
Also Catch if Cells is null and return null

Lennix committed:
Changes to Startup

Catch errors on Startup and return true/false
StartDispatcher added in HearthbeatDispatcher to start after Startup

Lennix committed:
Forgot .ToByteString()

AeonLucid commented on POGOLib/Net/Session.cs:
```diff
@@ -31,12 +31,20 @@ internal Session(AccessToken accessToken, string password, GeoCoordinate geoCoor
             _heartbeat = new HeartbeatDispatcher(this);
         }
 
+        public bool Startup()
```
What happens when startup is called after startup already happened?

Lennix replied:
It sends the requests again

AeonLucid commented on POGOLib/Net/RPCClient.cs:
```diff
@@ -69,6 +69,68 @@ internal RpcClient(Session session)
         /// <summary>
```
Wrong docs for this method.

AeonLucid commented on POGOLib/Net/RPCClient.cs:
```diff
@@ -69,6 +69,68 @@ internal RpcClient(Session session)
         /// <summary>
         /// It is not recommended to call this. Map objects will update automatically and fire the <see cref="Map.Update"/> event.
         /// </summary>
+        public bool Startup()
```
Change to internal as it should only be called from the session startup.

AeonLucid commented:
Hey, looks pretty good. I've added some notes, do you think you could make those changes? Will be happy to merge after.

AeonLucid commented:
Also, could you maybe update the demo to reflect your changes regarding `Startup()`? 

I am currently on holiday so I don't have access to stuff to code, just my phone. :(

Lennix commented:
Sure, will do

Lennix committed:
Minor changes to RpcClient.Startup

Updated summary, removed LevelUpRewards (turns out, its not always sent), changed to internal

Lennix committed:
Missing word

Lennix committed:
Save settings for BaseBagItems/Pokemon in Inventory

Lennix committed:
Merge branch 'master' of https://github.com/Lennix/POGOLib

Lennix committed:
Added Startup to demo

Lennix committed:
Technically its minDistance

Not sure about the internal interval, client might sometimes send
heartbeat after more than minDistance meter

Lennix commented:
Merged via PullRequest #18
